include/flatbuffers/flatc.h

```diff
@@ -22,11 +25,18 @@
 #include "flatbuffers/idl.h"
 #include "flatbuffers/util.h"
 
-#ifndef FLATC_H_
-#  define FLATC_H_
-
 namespace flatbuffers {
 
+class FlatCompilerLogger {
+  FlatCompilerLogger() {}
+public:
+  // There is no way to notify about an error from a code-generator
+  // except `assert` call.
+  // TODO: re-design interface to code generators.
+  static void Warn(const std::string &warn);
```

Comment:
If these are static, why aren't they simple extern functions?

Reply:
They should be extern functions, I will fix it.

The compilers have a non-extensible interface:
https://github.com/google/flatbuffers/blob/eddebec1b6136a3248c99aff1bcb4c3012a766be/include/flatbuffers/flatc.h#L34-L40
I've tried to replace the pack of args (parser, path, file_name) by reference to struct or class object with additional logging methods but refused due to too big changes in the code.